src/coder/decode.js

```diff
@@ -33,6 +33,10 @@ class Decoder {
     const msgs = []
 
     while (true) {
+      if (!this._buffer.length) {
+        // after consuming the whole length, _buffer has 0 length so don't want to bother varint
+        break
+      }
```

Comment:
```suggestion
    while (this._buffer.length) {
```

Suggested refactor - this way there's no new branch added so no need to add a test to ensure it's being hit.

Please can you check this against your CPU profiling to ensure it gives you the same performance improvement?

Reply:
yeah this is great, should be the same with no new branch added 👍 . Thanks @achingbrain !